### PR TITLE
Add Address_used check for CoinVerifier

### PR DESCRIPTION
**Pull request by Szpoti**

### CODE REVIEW

**Review by adamPetho — APPROVED** [review]
LGTM

File: WalletWasabi/WabiSabi/Backend/Banning/CoinVerifier.cs [hunk 1]
```diff
@@ -148,8 +149,8 @@ private bool CheckForFlags(ApiResponseItem response)
 		}
 
 		shouldBan = flagIds.Any(id => WabiSabiConfig.RiskFlags.Contains(id));
-
-		return shouldBan;
+		shouldRemove = !response.Report_info_section.Address_used;
```
Comment (adamPetho) [inline WalletWasabi/WabiSabi/Backend/Banning/CoinVerifier.cs]: Quick fix could be this
```suggestion
		shouldRemove = shouldBan || !response.Report_info_section.Address_used;
```

Or make sure the record is updated properly by changing the property setters.

**Comment by Szpoti** [comment]
@molnard I went with Adam's suggestion above.
It makes more sense to set the value where we set everything based on the response we get back, rather then have a setter (or do it in constructor) to set it afterwards, kinda doublechecking ourselves if the value we set is incorrect or not. As we don't touch neither `shouldRemove` nor `shouldBan` outside, this makes the most sense to me.

**Review by molnard — APPROVED** [review]
LGTM

**Comment by molnard** [comment]
Sorry I made a mistake here, merged accidentally. Open a PR in the main repo with the change and then reset your master to upstream/master.

**Pull request merged**